pkgs/development/python-modules/etebase/default.nix

```diff
@@ -1,56 +1,59 @@
-{ lib, stdenv
-, wheel
+{ lib
+, stdenv
+, fetchFromGitHub
+, buildPythonPackage
 , rustPlatform
-, pipInstallHook
+, pkg-config
+, rustfmt
 , setuptools-rust
-, python
-, msgpack
-, requests
 , openssl
-, perl
-, rustfmt
-, fetchFromGitHub
 , Security
+, msgpack
 }:
 
-rustPlatform.buildRustPackage rec {
+buildPythonPackage rec {
   pname = "etebase";
-  version = "0.31.1";
+  version = "0.31.2";
 
   src = fetchFromGitHub {
     owner = "etesync";
     repo = "etebase-py";
     rev = "v${version}";
-    sha256 = "163iw64l8lwawf84qswcjsq9p8qddv9ysjrr3dzqpqxb2yb0sy39";
+    hash = "sha256-enGmfXW8eV6FgdHfJqXr1orAsGbxDz9xUY6T706sf5U=";
   };
 
-  cargoSha256 = "0w8ypl6kj1mf6ahbdiwbd4jw6ldxdaig47zwk91jjsww5lbyx4lf";
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src;
+    name = "${pname}-${version}";
+    hash = "sha256-4eJvFf6aY+DYkrYgam5Ok9941PX4uQOmtRznEY0+1TE=";
+  };
+
+  format = "pyproject";
 
   nativeBuildInputs = [
+    pkg-config
     rustfmt
-    perl
-    openssl
-    pipInstallHook
     setuptools-rust
-    wheel
-  ];
+  ] ++ (with rustPlatform; [
+    cargoSetupHook
+    rust.cargo
+    rust.rustc
+  ]);
 
-  buildInputs = lib.optionals stdenv.isDarwin [ Security ];
+  buildInputs = [ openssl ] ++ lib.optionals stdenv.isDarwin [ Security ];
```

Comment:
Does this link to openssl and Security in darwin?

Reply:
I don't know, I don't have a Mac. I also don't know if the Security framework an OpenSSL are overlapping or not. From the API docs, it doesn't seem so. But hard to tell without a Mac.

Reply:
The package won't build without it. Not sure if the package does everything correct because the dylib is not linked against it but I don't wanna patch that so it will stay like this.